File: TransCoder_tokenized_test_set_functions/python/FIND_THE_FIRST_MISSING_NUMBER.py
def findFirstMissing ( array, start, end ) :
    if ( start > end ) :
        return end + 1
    if ( start != array [ start ] ) :
        return start;
        
    mid = int ( ( start + end ) / 2 )
    if ( array [ mid ] == mid ) :
        return findFirstMissing ( array, mid + 1, end )
    return findFirstMissing ( array, start, mid )

File: results/py/FIND_THE_FIRST_MISSING_NUMBER.cpp
int f_filled ( int * array , int start , int end ) {
  if ( ( start > end ) || ( end < start ) ) return end + 1 ;
  if ( ( start != array [ start ] ) && ( start != array [ start ] ) ) return start ;
  ;
  int mid = ( int ) ( ( start + end ) / 2 ) ;
  if ( ( array [ mid ] == mid ) || ( array [ mid ] == mid ) ) return f_filled ( array , mid + 1 , end ) ;
  return f_filled ( array , start , mid ) ;
}
